src/main/java/gyro/google/dns/PolicyResource.java

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright 2019, Perfect Sense, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.google.dns;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.google.api.client.util.Data;
+import com.google.api.services.dns.Dns;
+import com.google.api.services.dns.model.PoliciesPatchResponse;
+import com.google.api.services.dns.model.Policy;
+import com.google.api.services.dns.model.PolicyAlternativeNameServerConfig;
+import com.google.api.services.dns.model.PolicyNetwork;
+import gyro.core.GyroException;
+import gyro.core.GyroUI;
+import gyro.core.Type;
+import gyro.core.resource.Output;
+import gyro.core.resource.Resource;
+import gyro.core.resource.Updatable;
+import gyro.core.scope.State;
+import gyro.core.validation.Required;
+import gyro.google.Copyable;
+import gyro.google.GoogleResource;
+
+/**
+ * Creates a DNS Policy.
+ *
+ * Example
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *     google::dns-policy dns-policy-example
+ *         name: "dns-policy-example"
+ *         description: "DNS Policy Example"
+ *         network
+ *             network: $(google::compute-network network-example)
+ *         end
+ *     end
+ */
+@Type("dns-policy")
+public class PolicyResource extends GoogleResource implements Copyable<Policy> {
+
+    private DnsPolicyAlternativeNameServerConfig alternativeNameServerConfig;
+
+    private String description;
+
+    private Boolean enableInboundForwarding;
+
+    private Boolean enableLogging;
+
+    private String id;
+
+    private String name;
+
+    private List<DnsPolicyNetwork> network;
+
+    /**
+     * Sets an alternative name server for the associated networks. When specified, all DNS queries are forwarded to a name server that you choose. Names such as .internal are not available when an alternative name server is specified. The value may be ``null``.
```

Comment:
This doc is to help a user configure the value for the field. As someone cannot set ``null`` in the config this portion is not necessary. Unless a field is required, it is optional and so if not configured by a user, will behave accordingly.

Also by convention we try to reword the docs a bit, and not directly use what  is there in the API.
```suggestion
     * Sets an alternative name server for the associated networks. When specified, all DNS queries are forwarded to a name server that you choose. Names such as .internal are not available when an alternative name server is specified.
```